src/services/SupabaseDataService.ts

```diff
@@ -247,50 +248,89 @@ export class SupabaseDataService extends BaseSupabaseService<KyselyDataDatabase>
     tokenIds: string[];
     chainId: number;
   }) {
-    console.log("[marketplace-api] Validating orders by token IDs", tokenIds);
+    console.log(
+      "[marketplace-api] Validating orders by token IDs",
+      tokenIds,
+      "on chain",
+      chainId,
+    );
     const ordersToUpdate: {
       id: string;
       invalidated: boolean;
       validator_codes: OrderValidatorCode[];
     }[] = [];
-    for (const tokenId of tokenIds) {
-      // Fetch all orders for token ID from database
-      const { data: matchingOrders, error } = await this.getOrdersByTokenId({
-        tokenId,
-        chainId,
-      });
-
-      if (error) {
+    const getOrdersResults = await Promise.all(
+      tokenIds.map(async (tokenId) =>
+        this.getOrdersByTokenId({
+          tokenId,
+          chainId,
+        }),
+      ),
+    );
+
+    if (getOrdersResults.some((res) => res.error)) {
+      throw new Error(
```

Comment:
How are thrown errors handled? Is Sentry monitoring the cron job?

Reply:
Errors are caught in `cron/OrderValidation.ts`, and then logged using `console.error()`, which should be picked up by Sentry as it's running in the same process.